lxd/container_lxc.go

```diff
@@ -365,15 +365,47 @@ func (c *containerLXC) initLXC() error {
 		}
 	}
 
-	// Setup limits
-	limitMemory, ok := c.expandedConfig["limits.memory"]
-	if ok && limitMemory != "" {
+	// Memory limits
+	limitMemory := c.expandedConfig["limits.memory"]
+	if limitMemory != "" {
 		err = lxcSetConfigItem(cc, "lxc.cgroup.memory.limit_in_bytes", limitMemory)
 		if err != nil {
 			return err
 		}
 	}
 
+	// CPU limits
+	cpuPriority := c.expandedConfig["limits.cpu.priority"]
+	cpuAllowance := c.expandedConfig["limits.cpu.allowance"]
+
+	if cpuPriority != "" || cpuAllowance != "" {
+		cpuShares, cpuCfsQuota, cpuCfsPeriod, err := deviceParseCPU(cpuAllowance, cpuPriority)
+		if err != nil {
+			return err
+		}
+
+		if cpuShares != "1024" {
+			err = lxcSetConfigItem(cc, "lxc.cgroup.cpu.shares", cpuShares)
```

Comment:
How do I clear this value? e.g. if I set cpuAlllowance and cpuPriority to something so that cpuShares is != 1024, and then I set them back to whatever their defaults are, will this value get unset?